File: output/ChatGPT/codenet/C++/Python/s189738089.py
dp = [float('inf')] * 101000
dp[0] = 0

N, K = map(int, input().split())
h = list(map(int, input().split()))

for i in range(N):
    for j in range(1, K + 1):
        dp[i + j] = min(dp[i + j], dp[i] + abs(h[i] - h[i + j]))

print(dp[N - 1])

File: output/ChatGPT/codenet/C++/Java/s189738089.java
import java.util.Scanner;
import java.util.Arrays;

public class s189738089 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int K = sc.nextInt();
        int[] h = new int[N];
        for (int i = 0; i < N; i++) {
            h[i] = sc.nextInt();
        }

        int[] dp = new int[101000];
        Arrays.fill(dp, Integer.MAX_VALUE);
        dp[0] = 0;

        for (int i = 0; i < N; i++) {
            for (int j = 1; j <= K; j++) {
                if (i + j < N) {
                    dp[i + j] = Math.min(dp[i + j], dp[i] + Math.abs(h[i] - h[i + j]));
                }
            }
        }

        System.out.println(dp[N - 1]);
    }
}